source/development_process/index.rst

```diff
@@ -0,0 +1,596 @@
+
+Developers must always use a fully qualified push command when force pushing,
+naming both the git remote and branch. To force push to a branch named
+``branch`` run::
+	git push --force origin branch
+
+Do NOT run::
+	git push --force
+
+Resolving Merge Conflicts
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Hopefully I'll get one in this document so I can add it to this section
+
+Rebasing and Squashing Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Often a developer will check out a new branch and while they are working on the
+branch different changes will be merged into develop. To pick up changes on
+develop, check out the working branch and run the following:
+
+.. code::shell
+	$ git checkout fancy-changes
+	$ git rebase develop
+
+This will add all of the changes merged into develop since the branches
+diverged onto ``fancy-new-changes``.
+
+Sometimes a series of commits should be combined into one large commit. This
+can be useful when there were many "work in progress" commits which do not need
+to clutter the git history. This is called squashing commits. First, find the
+oldest commit hash which should be squashed using ``git log``, in this example
+``abcde12345``. Then run:
+
+.. code::shell
+	$ git rebase -i abcde12345 # i stands for interactive
+
+Git will open up the editor and provide detailed instructions on how to choose
+which commits to keep or combine.
+
+Cherry Picking
+~~~~~~~~~~~~~~
+Sometimes it will be necessary to move several commits from one branch to
+another. This can be achieved simply by using ``git cherry-pick``. First, check
+out the branch which the commits will be moved to. Then, find the hash
+representing the commit using ``git log --all``, for example ``abcde12345``,
+and run:
+
+.. code::shell
+	$ git cherry-pick abcde2345
+
+This will add commit ``abcde12345`` to the current branch.
+
+
+When Disaster Strikes
+~~~~~~~~~~~~~~~~~~~~~
+On occasion disaster will strike, and it will appear that all has been lost. It
+is important not to panic, such mistakes can often be resolved with ``git
+reflog``. As long as the ``.git`` folder is intact, git keeps a log of the
+the changes made to the repository. Running ``git reflog`` will show a summary
+of recent actions on the repository. Choose the hash representing the state
+which should be restored to, for instance ``absde12345`` and run ``git reset
+abcde12345``.
+
+
+Common Python Errors and How to Fix Them
+----------------------------------------
+- I can't think of any...
+
+Writing the Chef Cookbook
+-------------------------
+
+- Pretty much the same as the rest of the project
+- Setting up test kitchen (link to updated internal chef docs)
+- Create the git repo
+- Create the issue tracker
+- Create the project skeleton with Berks
+
+Writing Chef Tests
+------------------
+
+*Tests are awesome!* You should always write tests! Bad things can happen if
+you don't write tests. Here's the toolset we use to write tests for Chef:
```

Comment:
@iankronquist is going to complain about your usage of "you" here, there, and everywhere.